v3/src/components/data-display/models/data-configuration-model.ts

```diff
@@ -446,35 +450,33 @@ export const DataConfigurationModel = types
       )
     }
   }))
-  .extend(self => {
-    // TODO: This is a hack to get around the fact that MST doesn't seem to cache this as expected
-    // when implemented as simple view.
-    let quantileScale: ScaleQuantile<string> | undefined = undefined
-    let previousLowAttributeColor: string | undefined
-    let previousHighAttributeColor: string | undefined
-
-    return {
-      views: {
-        get legendQuantileScale() {
-          if (
-            !quantileScale ||
-            previousLowAttributeColor !== self.lowColor ||
-            previousHighAttributeColor !== self.highColor
-          ) {
-            previousLowAttributeColor = self.lowColor
-            previousHighAttributeColor = self.highColor
-            quantileScale = scaleQuantile(self.numericValuesForAttrRole('legend'), self.quantileScaleColors)
-          }
-          return quantileScale
-        },
-      },
-      actions: {
-        invalidateQuantileScale() {
-          quantileScale = undefined
-        }
+  .views(self => ({
+    get legendQuantileScale() {
+      /**
+       *  Adjust the value range displayed by the legend based on the data configuration model's properties:
+       *  1. If all cases are hidden, the legend displays no range.
+       *  2. If `displayOnlySelectedCases` is true and not all cases are visible, the legend displays the range of all
+       *     cases, both hidden and visible.
+       *  3. Otherwise, the legend displays the range of only the visible cases.
+       *
+       *  TODO: When `displayOnlySelectedCases` is true and all visible cases have the exact same value for the legend
+       *  attribute, the legend should only reflect the values of the case(s) shown.
+       */
+      const allCasesCount = self.dataset?.items.length ?? 0
+      const hiddenCasesCount = self.hiddenCases.length ?? 0
+      const allCasesHidden = hiddenCasesCount === allCasesCount
+      let values: number[]
+      if (allCasesHidden) {
+        values = []
+      } else if (self.displayOnlySelectedCases && hiddenCasesCount > 0) {
+        const attribute = self.dataset?.attrFromID(self.attributeID("legend"))
+        values = attribute?.numValues ?? []
```

Comment:
If I'm right about the above, then this would be something like:
```suggestion
      } else if (self.displayOnlySelectedCases && self.unselectedCases.length > 0) {
        // determine the set of values that are not hidden by the user regardless of selection status
        // const attribute = self.dataset?.attrFromID(self.attributeID("legend"))
        // values = attribute?.numValues ?? []
```